python/tvm/contrib/pipeline_executor.py

```diff
@@ -0,0 +1,559 @@
+        """Create graph_executor list and return string format config.
+
+        Parameters
+        ----------
+
+        pipeline_mods : List[IRModule]
+          list of IRModule
+
+        mod_config : Dict[int, Dict[str, Any]]
+            modules and modules dependency configuration informaiton.
+
+        Returns
+        -------
+        mods : List[GraphModule]
+            Runtime graph module.
+
+        mod_config : str
+            mods configuration
+        """
+
+        mods = []
+        for pipeline_mod in pipeline_mods:
+            mod = graph_executor.GraphModule(
+                pipeline_mod["default"](pipeline_mods[pipeline_mod]["dev"])
+            )
+            mods.append(mod.module)
+
+        return mods, json.dumps(mod_config)
+
+
+class PipelineConfig(object):
+    """The wrapper of each module to be pipelined. The wrapper mainly includes the
+    module itself as well as the binding that represents the connections of this
+    module's inputs and outputs to other modules.
+    """
+
+    class ModuleWrapper:
+        """The class use use to represent Module and storage module index and
+        Binding information.
+        """
+
+        class Binding:
+            """The class that use to storage module connection information.
+               There are 2 types Binding Input:1 Output:2
+            Parameters
+            ----------
+
+            owner : ModuleWrapper
+                The class that own this interface, in such class there are
+                Module information like index, module name
+
+            io_type : str
+                The type of this binding. It can be either "input" or "output".
+
+            name : str/integer
+                Binding name, for input that is string for example "data0"
+                for output that is integer for example 0.
+            """
+
+            def __init__(self, owner, stype, name, data_type=None):
+                self.io_owner = owner
+                self.io_type = stype
+                self.name = str(name)
+                # These item that have dependency relation with self
+                self.bindings = []
+                # The item that self depend
+                self.parents = []
+
+                self.data_type = data_type
+
+            def get_name(self):
+                """get owner name and self name"""
+                owner_name = ""
+                if isinstance(self.io_owner, PipelineConfig.ModuleWrapper):
+                    owner_name = self.io_owner.name
+
+                return owner_name, self.name
+
+            def get_owner_indx(self):
+                """return index if owner is ModuleWrapper, if not return 0"""
+                if isinstance(self.io_owner, PipelineConfig.ModuleWrapper):
+                    return self.io_owner.index
+
+                # if not ModuleWrapper then owner is PipelineConfig, return 0
+                # to identify this is global interface
+                return 0
```

Comment:
I still feel use 0 to indicate global interface is not clear enough. I read through the use cases, you only use it for `index != 0`, so it should be fine to just make this function to be

```
def is_global_interface(self):
    return not isinstance(self.io_owner, PipelineConfig.ModuleWrapper)
```

btw, please change all `xx_indx`  to `xx_index` or `xx_idx`.

Reply:
fixed.